statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,858 @@
+                               True, False, False] + [False] * m)
+                func = func_dict[(None, None)]
+            p[:] = [init_alpha, init_beta, init_gamma, l0, b0, init_phi] + s0
+
+            # txi [alpha, beta, gamma, l0, b0, phi, s0,..,s_(m-1)]
+            # Have a quick look in the region for a good starting place for alpha etc.
+            # using guestimates for the levels
+            txi = xi & np.array(
+                [True, True, True, False, False, True] + [False] * m)
+            bounds = np.array([(0.0, 1.0), (0.0, 1.0), (0.0, 1.0),
+                               (0.0, None), (0.0, None), (0.0, 1.0)] + [(None, None), ] * m)
+            res = brute(func, bounds[txi], (txi, p, y, l, b, s, m, self.nobs, max_seen),
+                        Ns=20, full_output=True, finish=None)
+            (p[txi], max_seen, grid, Jout) = res
+            [alpha, beta, gamma, l0, b0, phi] = p[:6]
+            s0 = p[6:]
+            #bounds = np.array([(0.0,1.0),(0.0,1.0),(0.0,1.0),(0.0,None),(0.0,None),(0.8,1.0)] + [(None,None),]*m)
+            if use_basinhopping:
+                # Take a deeper look in the local minimum we are in to find the best
+                # solution to parameters, maybe hop around to try escape the local
+                # minimum we may be in.
+                res = basinhopping(func, p[xi], minimizer_kwargs={'args': (
+                    xi, p, y, l, b, s, m, self.nobs, max_seen), 'bounds': bounds[xi]}, stepsize=0.01)
+            else:
+                # Take a deeper look in the local minimum we are in to find the best
+                # solution to parameters
+                res = minimize(func, p[xi], args=(
+                    xi, p, y, l, b, s, m, self.nobs, max_seen), bounds=bounds[xi])
+            p[xi] = res.x
+            [alpha, beta, gamma, l0, b0, phi] = p[:6]
+            s0 = p[6:]
+            opt = res
+        hwfit = self._predict(h=0, alpha=alpha, beta=beta, gamma=gamma, phi=phi, l0=l0,
+                              b0=b0, s0=s0, use_boxcox=use_boxcox, lamda=lamda, remove_bias=remove_bias)
+        hwfit.opt = opt
+        return hwfit
+
+    def _predict(self, h=None, alpha=None, beta=None, gamma=None, l0=None, b0=None,
+                 phi=None, s0=None, use_boxcox=None, lamda=None, remove_bias=None):
+        """
+        Helper prediction function
+
+        Parameters
+        ----------
+        h : int, optional
+            The number of time steps to forecast ahead.
+        """
+        # Start in sample and out of sample predictions
+        data = self.endog
+        damped = self.damped
+        seasoning = self.seasoning
+        trending = self.trending
+        trend = self.trend
+        seasonal = self.seasonal
+        m = self.seasonal_periods
+        phi = phi if damped else 1.0
+        if use_boxcox == 'log':
+            lamda = 0.0
+            y = boxcox(data, 0.0)
+        elif use_boxcox:
```

Comment:
this is missing the case when lamda is already a float.

Reply:
done